lib/operations/trigonometric.js

```diff
@@ -9,85 +10,159 @@ var utils = require('./utils')
 var misc = require('./misc')
 var algebra = require('./algebra')
 var arithmetic = require('./arithmetic')
+var division = require('./division')
 
 var trigonometric = {}
 
-trigonometric.cos = function (x) {
-  var rlo, rhi
-  if (utils.isEmpty(x)) { return constants.EMPTY }
-
-  // cos works with positive intervals only
-  if (x.lo < 0) {
-    var mult = 1e7
-    x.lo += 2 * Math.PI * mult
-    x.hi += 2 * Math.PI * mult
-  }
+// Ideas taken from ValidatedNumerics.jl and
+// Validated Numerics: A Short Introduction to Rigorous Computations, W. Tucker, 
+// Princeton University Press (2010), Chapter 3.
+
+// First we'll create a function that'll decide in which quadrant of the trig. circle our interval belongs.
+trigonometric.find_quadrant = function (x) {
+	var temp = division.nonZero(Interval(x), constants.PI_HALF)
+//	if (utils.isNotNegative(temp) == false){ 		// This trick doesn't work for all intervals
+//		var temp = arithmetic.negative(temp)		// that's why I had to change the signs manually inside the sin(x) func.
+//   }
+	return Interval(Math.floor(temp.lo), Math.floor(temp.hi))
+}
 
-  var pi2 = constants.PI_TWICE
-  var t = algebra.fmod(x, pi2)
-  if (misc.width(t) >= pi2.lo) {
-    return Interval(-1, 1)
-  }
+/**
+ * Returns the sine function of an interval x.
+ * @param {Interval} x
+ * @returns {Interval}
+ */
+trigonometric.sin = function (x) {
```

Comment:
`sin` and `cos` are special as you've seen, since the logic is kinda straightforward for Intervals where `lo >= 0` my idea was to move all the values to `k * 2 PI` times to the right if they were negative, I'd rather keep this instead of doing division as it propagates the values in a higher magnitude than doing simple multiplication